packages/angular/ssr/src/app.ts

```diff
@@ -124,6 +131,66 @@ export class AngularServerApp {
     ]);
   }
 
+  /**
+   * Serves a static, prerendered page for the incoming request if available.
+   *
+   * This method attempts to match the incoming request URL with a prerendered route in the application.
+   * If the route is found and is configured for prerendering, the corresponding static asset is retrieved
+   * and returned as a `Response` object with the appropriate HTTP headers.
+   *
+   * @param request - The incoming HTTP request for a potential static page.
+   * @returns A promise that resolves to a `Response` object containing the prerendered page content
+   * if available, or `null` if the request does not match a prerendered route or asset.
+   */
+  async serve(request: Request): Promise<Response | null> {
+    const url = stripIndexHtmlFromURL(new URL(request.url));
```

Comment:
Question: Does this imply that we're serving `/foo.html` (and presumably the app router would perform a client-side nav to `/foo`)?

Would it be better to 301/302 redirect to `/foo`?

Reply:
No, the stripping of the `index.html` is simply done for matching the route. No redirections or change in URL is involved here.